### ISSUE TITLE
=app-editors/leksah-server-0.15.2.0-r2 fails at setup configure 

### ISSUE THREAD

**Issue by elprawn** [gentoo-haskell/gentoo-haskell#620]
Here is the full build log

 ```
* Package:    app-editors/leksah-server-0.15.2.0-r2
 * Repository: haskell
 * Maintainer: haskell@gentoo.org
 * USE:        abi_x86_64 amd64 elibc_glibc kernel_linux network-uri threaded userland_GNU
 * FEATURES:   preserve-libs sandbox userpriv usersandbox
>>> Unpacking source...
>>> Unpacking leksah-server-0.15.2.0.tar.gz to /var/tmp/portage/app-editors/leksah-server-0.15.2.0-r2/work
>>> Source unpacked in /var/tmp/portage/app-editors/leksah-server-0.15.2.0-r2/work
>>> Preparing source in /var/tmp/portage/app-editors/leksah-server-0.15.2.0-r2/work/leksah-server-0.15.2.0 ...
 * CHDEP: 'Cabal >=1.10.2.0 && <1.23' -> 'Cabal >=1.10.2.0'
 * CHDEP: 'transformers >=0.2.2.0 && <0.5' -> 'transformers >=0.2.2.0'
>>> Source prepared.
>>> Configuring source in /var/tmp/portage/app-editors/leksah-server-0.15.2.0-r2/work/leksah-server-0.15.2.0 ...
 * Using cabal-1.24.2.0.
 * Prepending /usr/lib64/ghc-8.0.2 to LD_LIBRARY_PATH
/usr/bin/ghc -package Cabal-1.24.2.0 --make /var/tmp/portage/app-editors/leksah-server-0.15.2.0-r2/work/leksah-server-0.15.2.0/Setup.lhs -j5 +RTS -A256M -qb0 -RTS -threaded -dynamic -o setup
[1 of 1] Compiling Main             ( /var/tmp/portage/app-editors/leksah-server-0.15.2.0-r2/work/leksah-server-0.15.2.0/Setup.lhs, /var/tmp/portage/app-editors/leksah-server-0.15.2.0-r2/work/leksah-server-0.15.2.0/Setup.o )
Linking setup ...
./setup configure --ghc --prefix=/usr --with-compiler=/usr/bin/ghc --with-hc-pkg=/usr/bin/ghc-pkg --prefix=/usr --libdir=/usr/lib64 --libsubdir=leksah-server-0.15.2.0/ghc-8.0.2 --datadir=/usr/share/ --datasubdir=leksah-server-0.15.2.0/ghc-8.0.2 --disable-tests --ghc-options=-j5 +RTS -A256M -qb0 -RTS --with-ar=x86_64-pc-linux-gnu-ar --ghc-option=-optc-O2 --ghc-option=-optc-pipe --ghc-option=-optc-march=ivybridge --ghc-option=-optl-Wl,-O1 --ghc-option=-optl-Wl,--as-needed --disable-executable-stripping --docdir=/usr/share/doc/leksah-server-0.15.2.0-r2 --verbose --enable-shared --enable-executable-dynamic --sysconfdir=/etc --disable-library-stripping --flags=network-uri --flags=threaded
Configuring leksah-server-0.15.2.0...
setup: Encountered missing dependencies:
base >=4.0.0.0 && <4.9,
bin-package-db -any,
directory >=1.0.0.2 && <1.3,
ghc >=7.0.4 && <7.11,
haddock-api ==2.16.*,
haddock-library >=1.2.0 && <1.3,
process >=1.1 && <1.3,
time >=1.1 && <1.6
 * ghc-pkg check: 'checking for other broken packages:'
 * ERROR: app-editors/leksah-server-0.15.2.0-r2::haskell failed (configure phase):
 *   setup configure failed
 * 
 * Call stack:
 *     ebuild.sh, line  115:  Called src_configure
 *   environment, line 2837:  Called haskell-cabal_src_configure '--flags=network-uri' '--flags=threaded'
 *   environment, line 1959:  Called cabal-configure '--flags=network-uri' '--flags=threaded'
 *   environment, line  564:  Called cabal-show-brokens-and-die 'setup configure failed'
 *   environment, line  662:  Called die
 * The specific snippet of code:
 *       die "$@"
 * 
 * If you need support, post the output of `emerge --info '=app-editors/leksah-server-0.15.2.0-r2::haskell'`,
 * the complete build log and the output of `emerge -pqv '=app-editors/leksah-server-0.15.2.0-r2::haskell'`.
 * The complete build log is located at '/var/tmp/portage/app-editors/leksah-server-0.15.2.0-r2/temp/build.log'.
 * The ebuild environment file is located at '/var/tmp/portage/app-editors/leksah-server-0.15.2.0-r2/temp/environment'.
 * Working directory: '/var/tmp/portage/app-editors/leksah-server-0.15.2.0-r2/work/leksah-server-0.15.2.0'
 * S: '/var/tmp/portage/app-editors/leksah-server-0.15.2.0-r2/work/leksah-server-0.15.2.0'
```

I am using dev-lang/ghc-8.0.2::haskell, dev-haskell/haddock-library-1.4.2::haskell and dev-haskell/haddock-api-2.17.3::haskell, which are fine according to the dependencies specified in the ebuild, but the build log seems to suggest the package needs different versions. I am not sure what the other dependencies that it complains about are.

**Comment by elprawn** [comment]
I uninstalled all of the packages I had install that were related to haskell, then added the following lines to /etc/portage/package.mask, based on the previously posted build log's complaints.

```
<dev-lang/ghc-7.0.4
>=dev-lang/ghc-7.11
<dev-haskell/haddock-api-2.16
>dev-haskell/haddock-api-2.16
<dev-haskell/haddock-library-1.2
>=dev-haskell/haddock-library-1.3
<dev-haskell/cabal-1.10.2
>=dev-haskell/cabal-1.23
<dev-haskell/transformers-0.2.2
>=dev-haskell/transformers-0.5
```

Now when I run `emerge -av leksah`, I get the following output


```
These are the packages that would be merged, in order:

Calculating dependencies... done!
[ebuild  N    #] dev-lang/ghc-8.0.2:0/8.0.2::haskell  USE="gmp profile -binary -doc -ghcbootstrap -ghcmakebinary" 0 KiB
[ebuild  N    ~] app-admin/haskell-updater-1.2.10::haskell  0 KiB
[ebuild  N    #] dev-haskell/cabal-1.24.2.0:0/1.24.2.0::haskell  USE="-profile {-test}" 0 KiB
[ebuild  N     ] dev-haskell/transformers-0.4.3.0:0/0.4.3.0::haskell  USE="-doc -hoogle -hscolour -profile" 0 KiB
[ebuild  N    ~] dev-haskell/stm-2.4.4.1:0/2.4.4.1::haskell  USE="-doc -hoogle -hscolour -profile" 0 KiB
[ebuild  N    ~] dev-haskell/network-2.6.3.1-r1:0/2.6.3.1::haskell  USE="-doc -hoogle -hscolour -profile {-test}" 0 KiB
[ebuild  N     ] dev-haskell/binary-0.7.6.1:0/0.7.6.1::haskell  USE="-doc -hoogle -hscolour -profile {-test}" 0 KiB
[ebuild  N    ~] dev-haskell/base-orphans-0.6:0/0.6::haskell  USE="-doc -hoogle -hscolour -profile {-test}" 0 KiB
[ebuild  N    ~] dev-haskell/random-1.1:0/1.1::haskell  USE="-doc -hoogle -hscolour -profile {-test}" 0 KiB
[ebuild  N     ] dev-haskell/old-locale-1.0.0.7:0/1.0.0.7::haskell  USE="-doc -hoogle -hscolour -profile" 0 KiB
[ebuild  N    ~] dev-haskell/utf8-string-1.0.1.1:0/1.0.1.1::haskell  USE="-doc -hoogle -hscolour -profile" 0 KiB
[ebuild  N    ~] dev-haskell/hxt-charproperties-9.2.0.1:0/9.2.0.1::haskell  USE="-doc -hoogle -hscolour -profile" 0 KiB
[ebuild  N    ~] dev-haskell/executable-path-0.0.3.1:0/0.0.3.1::haskell  USE="-doc -hoogle -hscolour -profile" 0 KiB
[ebuild  N     ] dev-haskell/strict-0.3.2::haskell  USE="-doc -hoogle -hscolour -profile" 0 KiB
[ebuild  N    ~] dev-haskell/base-compat-0.9.3:0/0.9.3::haskell  USE="-doc -hoogle -hscolour -profile {-test}" 0 KiB
[ebuild  N    ~] dev-haskell/bytestring-builder-0.10.8.1.0:0/0.10.8.1.0::haskell  USE="-profile" 0 KiB
[ebuild  N    ~] dev-haskell/fail-4.9.0.0:0/4.9.0.0::haskell  USE="-profile" 0 KiB
[ebuild  N    ~] dev-haskell/cabal-doctest-1.0.1:0/1.0.1::haskell  USE="-doc -hoogle -hscolour -profile" 0 KiB
[ebuild  N    ~] dev-haskell/ghcjs-codemirror-0.0.0.1:0/0.0.0.1::haskell  USE="-doc -hoogle -hscolour -profile" 0 KiB
[ebuild  N    ~] dev-haskell/split-0.2.3.1-r1:0/0.2.3.1::haskell  USE="-doc -hoogle -hscolour -profile {-test}" 0 KiB
[ebuild  N    ~] dev-haskell/ansi-terminal-0.6.2.3:0/0.6.2.3::haskell  USE="-doc -example -hoogle -hscolour -profile" 0 KiB
[ebuild  N    ~] dev-haskell/hscolour-1.24.1:0/1.24.1::haskell  USE="-doc -hoogle -profile" 0 KiB
[ebuild  N    ~] dev-haskell/refact-0.3.0.2:0/0.3.0.2::haskell  USE="-doc -hoogle -hscolour -profile" 0 KiB
[ebuild  N    ~] dev-haskell/cmdargs-0.10.17:0/0.10.17::haskell  USE="quotation -doc -hoogle -hscolour -profile -testprog" 0 KiB
[ebuild  N    ~] dev-haskell/syb-0.7:0/0.7::haskell  USE="-doc -hoogle -hscolour -profile {-test}" 0 KiB
[ebuild  N    ~] dev-haskell/clock-0.7.2:0/0.7.2::haskell  USE="-doc -hoogle -hscolour -profile {-test}" 0 KiB
[ebuild  N    ~] dev-haskell/zlib-0.6.1.2:0/0.6.1.2::haskell  USE="-doc -hoogle -hscolour -non-blocking-ffi -profile {-test}" 0 KiB
[ebuild  N     ] dev-haskell/xhtml-3000.2.1-r1:0/3000.2.1::haskell  USE="-doc -hoogle -hscolour -profile" 0 KiB
[ebuild  N     ] dev-haskell/ghc-paths-0.1.0.9::haskell  USE="-profile" 0 KiB
[ebuild  N    ~] dev-haskell/dlist-0.8.0.2:0/0.8.0.2::haskell  USE="-doc -hoogle -hscolour -profile {-test}" 0 KiB
[ebuild  N    ~] dev-haskell/haskell-lexer-1.0.1:0/1.0.1::haskell  USE="-doc -hoogle -hscolour -profile" 0 KiB
[ebuild  N    ~] dev-haskell/parallel-3.2.1.1:3/3.2.1.1::haskell  USE="-doc -hoogle -hscolour -profile" 0 KiB
[ebuild  N     ] dev-haskell/prelude-extras-0.4.0.3:0/0.4.0.3::haskell  USE="-doc -hoogle -hscolour -profile" 0 KiB
[ebuild  N     ] dev-haskell/text-1.2.2.1:0/1.2.2.1::haskell  USE="-doc -hoogle -hscolour -profile {-test}" 0 KiB
[ebuild  N     ] dev-haskell/mtl-2.2.1-r1:0/2.2.1::haskell  USE="-doc -hoogle -hscolour -profile" 0 KiB
[ebuild  N    ~] dev-haskell/primitive-0.6.1.0-r1:0/0.6.1.0::haskell  USE="-doc -hoogle -hscolour -profile {-test}" 0 KiB
[ebuild  N     ] dev-haskell/old-time-1.1.0.3-r1:0/1.1.0.3::haskell  USE="-doc -hoogle -hscolour -profile" 0 KiB
[ebuild  N    ~] dev-haskell/async-2.1.1.1:0/2.1.1.1::haskell  USE="-doc -hoogle -hscolour -profile {-test}" 0 KiB
[ebuild  N    ~] dev-haskell/hxt-unicode-9.0.2.4:0/9.0.2.4::haskell  USE="-doc -hoogle -hscolour -profile" 0 KiB
[ebuild  N    ~] dev-haskell/extra-1.5.1:2/1.5.1::haskell  USE="-doc -hoogle -hscolour -profile {-test}" 0 KiB
[ebuild  N    #] dev-haskell/haddock-library-1.4.2:0/1.4.2::haskell  USE="-doc -hoogle -hscolour -profile {-test}" 0 KiB
[ebuild  N    ~] dev-haskell/time-locale-compat-0.1.1.3:0/0.1.1.3::haskell  USE="-doc -hoogle -hscolour -profile" 0 KiB
[ebuild  N     ] dev-haskell/statevar-1.1.0.4:0/1.1.0.4::haskell  USE="-doc -hoogle -hscolour -profile" 0 KiB
[ebuild  N    ~] dev-haskell/transformers-compat-0.5.1.4:0/0.5.1.4::haskell  USE="-doc -hoogle -hscolour -profile" 0 KiB
[ebuild  N    ~] dev-haskell/parsec-3.1.11:0/3.1.11::haskell  USE="-doc -hoogle -hscolour -profile {-test}" 0 KiB
[ebuild  N    ~] dev-haskell/hashable-1.2.6.0:0/1.2.6.0::haskell  USE="-doc -examples -hoogle -hscolour -profile {-test}" CPU_FLAGS_X86="sse2 -sse4_1" 0 KiB
[ebuild  N    ~] dev-haskell/vector-0.11.0.0:0/0.11.0.0::haskell  USE="boundschecks -doc -hoogle -hscolour -internalchecks -profile {-test} -unsafechecks" 0 KiB
[ebuild  N    ~] dev-haskell/blaze-builder-0.4.0.2:0/0.4.0.2::haskell  USE="-doc -hoogle -hscolour -profile {-test}" 0 KiB
[ebuild  N     ] dev-haskell/regex-base-0.93.2-r1:0/0.93.2::haskell  USE="-doc -hoogle -hscolour -profile" 0 KiB
[ebuild  N     ] dev-haskell/happy-1.19.5::haskell  USE="-doc {-test}" 0 KiB
[ebuild  N    ~] dev-haskell/hslogger-1.2.10:0/1.2.10::haskell  USE="-doc -hoogle -hscolour -profile" 0 KiB
[ebuild  N    ~] dev-haskell/binary-shared-0.8.3:0/0.8.3::haskell  USE="-doc -hoogle -hscolour -profile" 0 KiB
[ebuild  N    ~] dev-haskell/polyparse-1.12:0/1.12::haskell  USE="-doc -hoogle -hscolour -profile" 0 KiB
[ebuild  N     ] dev-haskell/tf-random-0.5:0/0.5::haskell  USE="-profile" 0 KiB
[ebuild  N    #] dev-haskell/haddock-api-2.17.3:0/2.17.3::haskell  USE="-doc -hoogle -hscolour -profile {-test}" 0 KiB
[ebuild  N    ~] dev-haskell/tagged-0.8.5:0/0.8.5::haskell  USE="-doc -hoogle -hscolour -profile" 0 KiB
[ebuild  N    ~] dev-haskell/exceptions-0.8.3-r1:0/0.8.3::haskell  USE="-doc -hoogle -hscolour -profile {-test}" 0 KiB
[ebuild  N    ~] dev-haskell/unordered-containers-0.2.8.0:0/0.2.8.0::haskell  USE="-debug -doc -hoogle -hscolour -profile {-test}" 0 KiB
[ebuild  N     ] dev-haskell/transformers-base-0.4.4:0/0.4.4::haskell  USE="orphaninstances -doc -hoogle -hscolour -profile" 0 KiB
[ebuild  N    ~] dev-haskell/network-uri-2.6.1.0:0/2.6.1.0::haskell  USE="-doc -hoogle -hscolour -profile {-test}" 0 KiB
[ebuild  N    ~] dev-haskell/cpphs-1.20.5:0/1.20.5::haskell  USE="-doc -hoogle -hscolour -profile" 0 KiB
[ebuild  N    ~] dev-haskell/nats-1.1.1:0/1.1.1::haskell  USE="binary hashable template-haskell -profile" 0 KiB
[ebuild  N    ~] dev-haskell/quickcheck-2.8.2:2/2.8.2::haskell  USE="template_haskell -doc -hoogle -hscolour -profile {-test}" 0 KiB
[ebuild  N    ~] dev-haskell/regex-tdfa-1.2.2:0/1.2.2::haskell  USE="-doc -hoogle -hscolour -profile" 0 KiB
[ebuild  N    ~] dev-haskell/blaze-markup-0.7.1.1:0/0.7.1.1::haskell  USE="-doc -hoogle -hscolour -profile {-test}" 0 KiB
[ebuild  N    ~] dev-haskell/mmorph-1.0.9:0/1.0.9::haskell  USE="-doc -hoogle -hscolour -profile" 0 KiB
[ebuild  N    ~] dev-haskell/pretty-show-1.6.12:0/1.6.12::haskell  USE="-doc -hoogle -hscolour -profile" 0 KiB
[ebuild  N    ~] dev-haskell/hashtables-1.2.1.1:0/1.2.1.1::haskell  USE="unsafe-tricks -bounds-checking -debug -doc -hoogle -hscolour -portable -profile" CPU_FLAGS_X86="-sse4_2" 0 KiB
[ebuild  N    ~] dev-haskell/hxt-regex-xmlschema-9.2.0.3:0/9.2.0.3::haskell  USE="-doc -hoogle -hscolour -profile {-test}" 0 KiB
[ebuild  N    ~] dev-haskell/streaming-commons-0.1.17:0/0.1.17::haskell  USE="-doc -hoogle -hscolour -profile {-test}" 0 KiB
[ebuild  N    ~] dev-haskell/semigroups-0.18.3:0/0.18.3::haskell  USE="binary transformers -doc -hoogle -hscolour -profile" 0 KiB
[ebuild  N     ] dev-haskell/monad-control-1.0.1.0:0/1.0.1.0::haskell  USE="-doc -hoogle -hscolour -profile" 0 KiB
[ebuild  N    ~] dev-haskell/blaze-html-0.8.1.3:0/0.8.1.3::haskell  USE="-doc -hoogle -hscolour -profile {-test}" 0 KiB
[ebuild  N    ~] dev-haskell/haskell-src-exts-1.17.1:0/1.17.1::haskell  USE="-doc -hoogle -hscolour -profile {-test}" 0 KiB
[ebuild  N    ~] dev-haskell/regex-tdfa-text-1.0.0.3:0/1.0.0.3::haskell  USE="-doc -hoogle -hscolour -profile" 0 KiB
[ebuild  N    ~] dev-haskell/alex-3.2.1::haskell  USE="-doc {-test}" 0 KiB
[ebuild  N    ~] dev-haskell/hxt-9.3.1.16:0/9.3.1.16::haskell  USE="network-uri -doc -hoogle -hscolour -profile" 0 KiB
[ebuild  N    ~] dev-haskell/integer-logarithms-1.0.1:0/1.0.1::haskell  USE="-doc -hoogle -hscolour -profile {-test}" 0 KiB
[ebuild  N     ] dev-haskell/uniplate-1.6.12:0/1.6.12::haskell  USE="-doc -hoogle -hscolour -profile" 0 KiB
[ebuild  N    ~] dev-haskell/http-4000.3.6:0/4000.3.6::haskell  USE="network-uri -conduit10 -doc -hoogle -hscolour -profile {-test}" 0 KiB
[ebuild  N    ~] dev-haskell/reflection-2.1.2:0/2.1.2::haskell  USE="template-haskell -doc -hoogle -hscolour -profile" 0 KiB
[ebuild  N    ~] dev-haskell/distributive-0.5.2:0/0.5.2::haskell  USE="-doc -hoogle -hscolour -profile {-test}" 0 KiB
[ebuild  N    ~] dev-haskell/gtk2hs-buildtools-0.13.2.2:0/0.13.2.2::haskell  USE="closuresignals -profile" 0 KiB
[ebuild  N    ~] dev-haskell/void-0.7.2:0/0.7.2::haskell  USE="-doc -hoogle -hscolour -profile -safe" 0 KiB
[ebuild  N    ~] dev-haskell/scientific-0.3.4.12:0/0.3.4.12::haskell  USE="-doc -hoogle -hscolour -profile {-test}" 0 KiB
[ebuild  N    ~] dev-haskell/vcswrapper-0.1.5:0/0.1.5::haskell  USE="-doc -hoogle -hscolour -profile" 0 KiB
[ebuild  N    ~] dev-haskell/lifted-base-0.2.3.10:0/0.2.3.10::haskell  USE="-doc -hoogle -hscolour -profile {-test}" 0 KiB
[ebuild  N    ~] dev-haskell/hlint-1.9.35:0/1.9.35::haskell  USE="gpl -doc -emacs -hoogle -hscolour -profile" 0 KiB
[ebuild  N    ~] dev-haskell/contravariant-1.4:0/1.4::haskell  USE="semigroups statevar tagged -doc -hoogle -hscolour -profile -safe" 0 KiB
[ebuild  N    ~] dev-haskell/attoparsec-0.13.1.0:0/0.13.1.0::haskell  USE="-doc -hoogle -hscolour -profile {-test}" 0 KiB
[ebuild  N    ~] dev-haskell/resourcet-1.1.9:0/1.1.9::haskell  USE="-doc -hoogle -hscolour -profile {-test}" 0 KiB
[ebuild  N    ~] dev-haskell/comonad-5.0.1:0/5.0.1::haskell  USE="containers contravariant distributive test-doctests -doc -hoogle -hscolour -profile {-test}" 0 KiB
[ebuild  N    ~] dev-haskell/conduit-1.2.10:0/1.2.10::haskell  USE="-doc -hoogle -hscolour -profile {-test}" 0 KiB
[ebuild  N    ~] dev-haskell/vado-0.0.8:0/0.0.8::haskell  USE="-doc -hoogle -hscolour -profile {-test}" 0 KiB
[ebuild  N    ~] dev-haskell/aeson-1.0.2.1-r1:0/1.0.2.1::haskell  USE="-doc -hoogle -hscolour -profile {-test}" 0 KiB
[ebuild  N    ~] dev-haskell/bifunctors-5.4.2:0/5.4.2::haskell  USE="semigroups tagged -doc -hoogle -hscolour -profile {-test}" 0 KiB
[ebuild  N    ~] dev-haskell/shakespeare-2.0.13:0/2.0.13::haskell  USE="-doc -hoogle -hscolour -profile {-test} -test_coffee -test_export -test_roy" 0 KiB
[ebuild  N    ~] dev-haskell/conduit-extra-1.1.15:0/1.1.15::haskell  USE="-doc -hoogle -hscolour -profile {-test}" 0 KiB
[ebuild  N    ~] dev-haskell/semigroupoids-5.1:0/5.1::haskell  USE="comonad containers contravariant distributive tagged -doc -hoogle -hscolour -profile {-test}" 0 KiB
[ebuild  N    ~] dev-haskell/profunctors-5.2-r1:0/5.2::haskell  USE="-doc -hoogle -hscolour -profile" 0 KiB
[ebuild  N     ] dev-haskell/free-4.12.4:0/4.12.4::haskell  USE="-doc -hoogle -hscolour -profile" 0 KiB
[ebuild  N    ~] dev-haskell/adjunctions-4.3:0/4.3::haskell  USE="-doc -hoogle -hscolour -profile" 0 KiB
[ebuild  N    ~] dev-haskell/kan-extensions-5.0.1:0/5.0.1::haskell  USE="-doc -hoogle -hscolour -profile" 0 KiB
[ebuild  N    ~] dev-haskell/lens-4.14:0/4.14::haskell  USE="inlining test-doctests test-templates trustworthy -benchmark-uniplate -doc -dump-splices -hoogle -hscolour -j -old-inline-pragmas -profile -safe {-test}" 0 KiB
[ebuild  N    ~] dev-haskell/glib-0.13.4.1:0/0.13.4.1::haskell  USE="closure_signals -doc -hoogle -hscolour -profile" 0 KiB
[ebuild  N    ~] dev-haskell/cairo-0.13.3.1:0/0.13.3.1::haskell  USE="pdf postscript svg -doc -hoogle -hscolour -profile" 0 KiB
[ebuild  N    ~] dev-haskell/gio-0.13.3.1:0/0.13.3.1::haskell  USE="-doc -hoogle -hscolour -profile" 0 KiB
[ebuild  N    ~] dev-haskell/pango-0.13.3.1:0/0.13.3.1::haskell  USE="-doc -hoogle -hscolour -profile" 0 KiB
[ebuild  N    ~] dev-haskell/gtk3-0.14.6:0/0.14.6::haskell  USE="fmode-binary gio -doc -hoogle -hscolour -profile" 0 KiB
[ebuild  N    ~] dev-haskell/ltk-0.15.0.5-r1:0/0.15.0.5::haskell  USE="-doc -hoogle -hscolour -profile" 0 KiB
[ebuild  N    ~] dev-haskell/vcsgui-0.1.3.0:0/0.1.3.0::haskell  USE="gtk3 -doc -hoogle -hscolour -profile" 0 KiB
[ebuild  N    ~] dev-haskell/gtksourceview3-0.13.3.1:0/0.13.3.1::haskell  USE="-doc -hoogle -hscolour -profile" 0 KiB
[ebuild  N    ~] app-editors/leksah-server-0.15.2.0-r2:0/0.15.2.0::haskell  USE="network-uri threaded -doc -hoogle -hscolour -profile {-test}" 0 KiB
[ebuild  N    ~] dev-haskell/webkitgtk3-0.14.2.1:0/0.14.2.1::haskell  USE="-doc -hoogle -hscolour -profile" 0 KiB
[ebuild  N    ~] dev-haskell/webkitgtk3-javascriptcore-0.13.1.2:0/0.13.1.2::haskell  USE="-doc -hoogle -hscolour -profile" 0 KiB
[ebuild  N    ~] dev-haskell/ghcjs-dom-0.2.4.0:0/0.2.4.0::haskell  USE="gtk3 jsffi webkit -doc -hoogle -hscolour -jsc -profile" 0 KiB
[ebuild  N    ~] dev-haskell/jsaddle-0.3.0.3-r1:0/0.3.0.3::haskell  USE="gtk3 -doc -hoogle -hscolour -profile {-test}" 0 KiB
[ebuild  N    ~] app-editors/leksah-0.15.2.0-r3:0/0.15.2.0::haskell  USE="dyre network-uri threaded -doc -hoogle -hscolour -loc -profile {-test} -yi" 0 KiB

Total: 118 packages (118 new), Size of downloads: 0 KiB

WARNING: One or more updates/rebuilds have been skipped due to a dependency conflict:

dev-haskell/webkitgtk3-javascriptcore:0

  (dev-haskell/webkitgtk3-javascriptcore-0.14.2.1:0/0.14.2.1::haskell, ebuild scheduled for merge) conflicts with
    <dev-haskell/webkitgtk3-javascriptcore-0.14:= required by (dev-haskell/jsaddle-0.3.0.3-r1:0/0.3.0.3::haskell, ebuild scheduled for merge)
    ^                                      ^^^^  


The following mask changes are necessary to proceed:
 (see "package.unmask" in the portage(5) man page for more details)
# required by dev-haskell/gtk2hs-buildtools-0.13.2.2::haskell
# required by dev-haskell/webkitgtk3-javascriptcore-0.13.1.2::haskell
# required by app-editors/leksah-0.15.2.0-r3::haskell
# required by leksah (argument)
# /etc/portage/package.mask:
=dev-haskell/cabal-1.24.2.0
# required by app-editors/leksah-server-0.15.2.0-r2::haskell
# required by app-editors/leksah-0.15.2.0-r3::haskell
# required by leksah (argument)
# /etc/portage/package.mask:
=dev-haskell/haddock-api-2.17.3
# required by dev-haskell/haddock-api-2.17.3::haskell
# required by app-editors/leksah-server-0.15.2.0-r2::haskell
# required by app-editors/leksah-0.15.2.0-r3::haskell
# required by leksah (argument)
# /etc/portage/package.mask:
=dev-lang/ghc-8.0.2
# required by dev-haskell/haddock-api-2.17.3::haskell
# required by app-editors/leksah-server-0.15.2.0-r2::haskell
# required by app-editors/leksah-0.15.2.0-r3::haskell
# required by leksah (argument)
# /etc/portage/package.mask:
=dev-haskell/haddock-library-1.4.2

NOTE: The --autounmask-keep-masks option will prevent emerge
      from creating package.unmask or ** keyword changes.

Would you like to add these changes to your config files? [Yes/No] no
```

Now I am confused, because I am being told by emerge to unmask the versions that I masked in order to stop the build for leksah-server from failing.

Am I missing something? Thank you for your help.

**Comment by trofi** [comment]
I don't think leksah can be compiled with recent ghc. Let's see if leksah from git will be any better.

**Comment by elprawn** [comment]
I have updated the package.mask (some of it was specified incorrectly) and will try to see if I can get any further. It now looks like this

```
<=dev-lang/ghc-7.0.3
>=dev-lang/ghc-7.11
<=dev-haskell/haddock-api-2.15
>=dev-haskell/haddock-api-2.17
<=dev-haskell/haddock-library-1.1
>=dev-haskell/haddock-library-1.4
<=dev-haskell/transformers-0.2.1
>=dev-haskell/transformers-0.5
```

It is using ghc 7.10 now. If it doesn't work then I will try your suggestion of installing from git. Thank you.

**Comment by trofi** [comment]
I've pushed into `::haskell` overlay all dependencies git version of leksah has.
Also pushed git version of leksah as `leksah-0.16.9999.ebuild`. `leksah` builds and starts for me.

But at startup it seems to re-compile every single package I have installed.
Compilation happens in `~/.leksah-0.16/packageSources` (I guess for indexing purposes).

It might be quicker for you. But for me it will be 1855 packages. Thus I didn't see editor's window yet.

**Comment by elprawn** [comment]
I continue with what I said and leksah-server installed, but now dev-haskell/webkitgtk3-javascriptcore-0.13.1.2 complains about the cabal version.

I will resync the repo and emerge everything from scratch with nothing masked and see where I get. Thanks.

**Comment by elprawn** [comment]
I managed to install the new version on there. I added "/usr/lib64" as a path under which haskell sources for packages may be found (not sure if this is correct, but that is where all of the ones that came from the overlay are living). I also selected "Try to download and then build locally if that fails".

I pressed continue and after a few seconds the editor popped-up. Not sure if it's configured correctly or not, though, but it's fine for what I want, for now.

Thanks for your help.
